src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+goog.module('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * A class that processes events pushed to the data layer
+ * by constructing and sending Google Analytics events via
+ * Measurement Protocol.
+ * For its Google Analytic events, it generates a unique client ID
+ * and stores it in the long term storage model provided.
+ * @implements {EventProcessor}
+ */
+class GoogleAnalyticsEventProcessor {
+  /**
+   * @param {{
+   *     'api_secret': (string|undefined),
+   *     'measurement_id': (string|undefined),
+   *     'measurement_url': string,
+   *     'client_id_expires': number,
+   *     'automatic_params': !Object<string, boolean>,
+   * }=} optionsObject
+   */
+  constructor({
+      'api_secret': apiSecret,
+      'measurement_id': measurementId,
+      'measurement_url': measurementUrl = 'https://www.google-analytics.com/mp/collect',
+      'client_id_expires': clientIdExpires = Number.POSITIVE_INFINITY,
+      'automatic_params': userAutomaticParams = {},
+    } = {}) {
+    /**
+     * Parameters that are important to all events and will be searched for
+     * globally in the data model.
+     * @private @const {!Object<string, boolean>}
+     */
+    this.automaticParams_ = Object.assign({
+      'page_path': true,
+      'page_location': true,
+      'page_title': true,
+      'user_id': true,
+      'client_id': true,
+    }, userAutomaticParams);
+
+    /**
+     * Parameters that are to be set at the top level of the JSON
+     * POST body instead of as event parameters.
+     * @private @const {!Object<string,boolean>}
+     */
+    this.topLevelParams_ = {
```

Comment:
This feels like something that could be a top-level constant rather than a member of the class.